Hostap fails to compile because of warnings

There is a CI failure with this command:

```
west build -p -b nrf7002dk/nrf5340/cpuapp samples/net/wifi -T sample.net.wifi.nrf70dk -d ../build/wifi-sample/
```
Compile warning:
```
[3/475] Generating include/generated/zephyr/version.h
-- Zephyr version: 3.7.99 (zephyrproject/zephyr), build: v3.7.0-3649-g420fa472ab05
[459/475] Building C object zephyr/CMakeFiles/zephyr.dir/zephyrproject/modules/lib/hostap/src/drivers/driver_zephyr.c.obj
zephyrproject/modules/lib/hostap/src/drivers/driver_zephyr.c:988:13: warning: 'wpa_drv_zep_event_dfs_cac_finished' defined but not used [-Wunused-function]
  988 | static void wpa_drv_zep_event_dfs_cac_finished(struct zep_drv_if_ctx *if_ctx, union wpa_event_data *event)
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
zephyrproject/modules/lib/hostap/src/drivers/driver_zephyr.c:978:13: warning: 'wpa_drv_zep_event_dfs_cac_started' defined but not used [-Wunused-function]
  978 | static void wpa_drv_zep_event_dfs_cac_started(struct zep_drv_if_ctx *if_ctx, union wpa_event_data *event)
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[465/475] Building C object zephyr/CMakeFiles/zephyr.dir/zephyrproject/modules/lib/hostap/src/crypto/crypto_mbedtls_alt.c.obj
zephyrproject/modules/lib/hostap/src/crypto/crypto_mbedtls_alt.c:613:12: warning: 'hmac_prf256' defined but not used [-Wunused-function]
  613 | static int hmac_prf256(const u8 *key,
      |            ^~~~~~~~~~~
```

--- a/west.yml
+++ b/west.yml
@@ -259,7 +259,7 @@ manifest:
         - hal
     - name: hostap
       path: modules/lib/hostap
-      revision: 291e1cd0dfad83ec02826adf48dd6db69c3471ca
+      revision: 9896a2ea803ec62e0998c0e569f12af88537d647
     - name: libmetal
       revision: a6851ba6dba8c9e87d00c42f171a822f7a29639b
       path: modules/hal/libmetal

